### PR TITLE
[SYCL][NFC] Code clean up (phase 4) revealed by self build.

### PR DESCRIPTION
**Pull request by zahiraam**

### CODE REVIEW

File: llvm/tools/file-table-tform/file-table-tform.cpp [hunk 1]
```diff
@@ -235,7 +235,7 @@ struct TformCmd {
 
 #define CHECK_AND_EXIT(E)                                                      \
   {                                                                            \
-    Error LocE = std::move(E);                                                 \
+    Error LocE = E;                                                            \
```
Comment (kbobrovs) [inline llvm/tools/file-table-tform/file-table-tform.cpp:238]: aren't old and new variants equivalent? What problem does this change solve?

Comment (zahiraam) [reply llvm/tools/file-table-tform/file-table-tform.cpp:238]: This is the error I was trying to fix:
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(277,9): error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
        CHECK_AND_EXIT(Cmd.takeError());
        ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(238,18): note: expanded from macro 'CHECK_AND_EXIT'
    Error LocE = std::move(E);                                                 \
                 ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(288,5): error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
    CHECK_AND_EXIT(makeUserError("no inputs"));
    ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(238,18): note: expanded from macro 'CHECK_AND_EXIT'
    Error LocE = std::move(E);                                                 \
                 ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(303,5): error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
    CHECK_AND_EXIT(Table.takeError());
    ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(238,18): note: expanded from macro 'CHECK_AND_EXIT'
    Error LocE = std::move(E);                                                 \
                 ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(315,5): error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
    CHECK_AND_EXIT(createFileError(Output, EC));
    ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(238,18): note: expanded from macro 'CHECK_AND_EXIT'
    Error LocE = std::move(E);                                                 \
                 ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(319,5): error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
    CHECK_AND_EXIT(createFileError(Output, Out.error()));
    ^
D:/IUSERS/zahiraam/xmain_ws/xmain_self_build/llvm/llvm/tools/file-table-tform/file-table-tform.cpp(238,18): note: expanded from macro 'CHECK_AND_EXIT'
    Error LocE = std::move(E);                                                 \
                 ^
5 errors generated.

**Review by bader — APPROVED** [review]

File: llvm/tools/file-table-tform/file-table-tform.cpp (hunk 1 above)
Comment (kbobrovs) [reply llvm/tools/file-table-tform/file-table-tform.cpp:238]: Oh, right. I was looking at the call sites that are changed and could not see the temp. I should've looked to all other call sites instead.

**Review by kbobrovs — APPROVED** [review]
LGTM

**Comment by zahiraam** [comment]
@smaslov-intel Can you please review? Thanks.

**Review by smaslov-intel — APPROVED** [review]

**Comment by premanandrao** [comment]
@romanovvlad, is this something you can help merge?   Thanks!

**Pull request merged**